lib/pure/osproc.nim

```diff
@@ -747,6 +747,29 @@ elif not defined(useNimRtl):
     tags: [ExecIOEffect, ReadEnvEffect, ReadDirEffect, RootEffect], cdecl, gcsafe.}
   {.pop.}
 
+  proc posixFindExe*(exe: string): string {.
```

Comment:
Do we really need another proc for this? Wouldn't it be easier to add an option to ``findExe`` for this behaviour?

Reply:
I would still prefer an extending findExe instead of creating a separate proc.

Reply:
It shares almost no code with findExe, so I would just need to add another big `if`, which seems strange.

Reply:
`posixFindExe` also checks if the file is executable and raises EPERM when it's not. If you don't like this additional check (I've added it for compatibility with all other tools/languages that use execvp), I can merge both functions. Or add I can add `posixMode` flag to `findExe`.